usr/sbin/pkcsep11_migrate/pkcsep11_migrate.c

```diff
@@ -168,40 +162,46 @@ static int reencrypt(CK_SESSION_HANDLE session, CK_ULONG obj, CK_BYTE *old)
                   target);
 
     if (rc != CKR_OK || resp_len == 0) {
-        fprintf(stderr, "reencryption failed %lx %ld\n", rc, req_len);
+        fprintf(stderr, "reencryption failed: %lx %ld\n", rc, req_len);
         rc = -3;
         goto out;
     }
 
     if (_ep11a_internal_rv(resp, resp_len, &lrb, &rc) < 0) {
-        fprintf(stderr, "reencryption response malformed %lx\n", rc);
+        fprintf(stderr, "reencryption response malformed: %lx\n", rc);
         rc = -4;
         goto out;
     }
 
-    if (op_old->blob_size != lrb.pllen) {
-        fprintf(stderr, "reencryption blob size changed %lx %lx %lx %lx\n",
-                op_old->blob_size, lrb.pllen, resp_len, req_len);
+    if (rc != 0) {
+        fprintf(stderr, "reencryption failed: %lx\n", rc);
+        rc = -7;
+        goto out;
+    }
+
+    if (old_len != lrb.pllen) {
+        fprintf(stderr, "reencryption blob size changed: %lx %lx %lx %lx\n",
+                old_len, lrb.pllen, resp_len, req_len);
         rc = -5;
         goto out;
     }
 
-    memset(&op_new, 0, sizeof(op_new));
-    op_new.blob_id = op_old->blob_id;
-    op_new.blob_size = op_old->blob_size;
-    memcpy(op_new.blob, lrb.payload, op_new.blob_size);
+    memset(&blob, 0, sizeof(blob));
```

Comment:
what is &blob now ? the address of the pointer ?

Reply:
fixed. Actually &blob and just blob is the same in this context.